src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -516,19 +515,27 @@ interface RenderedResourceData {
 
 class RepositoryPaneActionRunner extends ActionRunner {
 
-	constructor(private getSelectedResources: () => (ISCMResource | IResourceNode<ISCMResource, ISCMResourceGroup>)[]) {
+	constructor(private getSelectedResources: () => (ISCMResource | ISCMResourceGroup | IResourceNode<ISCMResource, ISCMResourceGroup>)[]) {
 		super();
 	}
 
-	protected override async runAction(action: IAction, context: ISCMResource | IResourceNode<ISCMResource, ISCMResourceGroup>): Promise<any> {
+	protected override async runAction(action: IAction, context: ISCMResource | ISCMProvider | IResourceNode<ISCMResource, ISCMResourceGroup>): Promise<any> {
 		if (!(action instanceof MenuItemAction)) {
 			return super.runAction(action, context);
 		}
 
 		const selection = this.getSelectedResources();
-		const contextIsSelected = selection.some(s => s === context);
-		const actualContext = contextIsSelected ? selection : [context];
-		const args = flatten(actualContext.map(e => ResourceTree.isResourceNode(e) ? ResourceTree.collect(e) : [e]));
+		const contextAsProvider = <ISCMProvider>context;
+		const contextIsSelected = selection.some(s => s === context || (<ISCMResourceGroup>s)?.id === contextAsProvider?.id);
+		let actualContext = contextIsSelected ? selection : [context];
+		// When running action on selection, and the context is an ISCMProvider, convert the selection to providers.
+		if (contextIsSelected && contextAsProvider !== undefined) {
+			// This inserts the context at the beginning, and the rest of the selection as providers afterwards.
+			// This is done as some actions only run on the first argument, instead of all args.
+			actualContext = [context, ...selection.map(s => (<ISCMResourceGroup>s)?.provider).filter(p => p === undefined || p?.id === contextAsProvider.id)];
+		}
```

Comment:
Since `context` and `selection` have different types, we need to do some mapping here (see comments)